eventsource-client/src/client.rs

```diff
@@ -426,6 +490,35 @@ where
                             .set(State::WaitingToReconnect(delay(duration, "retrying")))
                     }
                 },
```

Comment:
I wasn't sure if usage of `std::io::Error` here constitutes an abuse of that type.